Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/CruiseControl.cs

```diff
@@ -0,0 +1,2277 @@
+                                                    controllerVolts -= step;
+                                                }
+                                            }
+                                            else
+                                            {
+                                                if (controllerVolts > -100)
+                                                {
+                                                    float step = 100 / DynamicBrakeFullRangeIncreaseTimeSeconds;
+                                                    step *= elapsedClockSeconds;
+                                                    controllerVolts -= step;
+                                                }
+                                            }
+                                        }
+                                        if (Locomotive.DynamicBrakePercent > -((AccelerationDemandMpSS - 0.05f) * 100))
+                                        {
+                                            if (controllerVolts < 0)
+                                            {
+                                                float step = 100 / DynamicBrakeFullRangeDecreaseTimeSeconds;
+                                                step *= elapsedClockSeconds;
+                                                controllerVolts += step;
+                                            }
+                                        }
+                                    }
+                                }
+                                else // use TrainBrake
+                                {
+                                    if (delta > -0.1)
+                                    {
+                                        if (!UseThrottle)
+                                            Locomotive.ThrottleController.SetPercent(100);
+                                        throttleIsZero = false;
+                                        maxForceN = 0;
+                                    }
+                                    else if (delta > -1)
+                                    {
+                                        if (!UseThrottle)
+                                            Locomotive.ThrottleController.SetPercent(0);
+                                        throttleIsZero = true;
+
+                                        brakePercent = TrainBrakeMinPercentValue - 3.0f + (-delta * 10);
+                                    }
+                                    else
+                                    {
+                                        Locomotive.TractiveForceN = 0;
+                                        if (!UseThrottle)
+                                            Locomotive.ThrottleController.SetPercent(0);
+                                        throttleIsZero = true;
+
+                                        if (RelativeAccelerationMpSS > -MaxDecelerationMpSS + 0.01f)
```

Comment:
I am lost with the variables relativeAcceleration, AccelerationDemandMpSS and RelativeAccelerationMpSS. All seem to have similar values. Please take a look at them.